### PR TITLE
Feat:level sys

### PR DESCRIPTION
**Pull request by lmangit**
- depend 
- https://github.com/Meland-Inc/egret-planet/pull/529 
- https://github.com/Meland-Inc/meland-unity-ui/pull/6
- https://github.com/Meland-Inc/meland-shared_unity_core/pull/4

### CODE REVIEW

File: Assets/Src/Module/MainPlayerRole/view/ComSkillInfoLogic.cs [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+        AddMessage();
+    }
+
+    public override void OnClose()
+    {
+        base.OnClose();
+        RemoveMessage();
+    }
+
+    private void AddUIEvent()
```
Comment (qqoo6789) [inline Assets/Src/Module/MainPlayerRole/view/ComSkillInfoLogic.cs:83]: 少了个对应的RemoveUIEvent

File: Assets/Src/Module/MainPlayerRole/view/ComSlotInfoLogic.cs [hunk 2]
```diff
@@ -14,31 +14,49 @@ public class ComSlotInfoLogic : FGUILogicCpt
...
+        AddMessage();
+    }
+
+    public override void OnClose()
+    {
+        base.OnClose();
+        RemoveMessage();
     }
 
     private void AddUIEvent()
```
Comment (qqoo6789) [inline Assets/Src/Module/MainPlayerRole/view/ComSlotInfoLogic.cs:62]: 同上少了 removeEvent

File: Assets/Src/Module/MainPlayerRole/RoleLevelModule.cs [hunk 3]
```diff
@@ -1,12 +1,145 @@
...
+    public const int MAX_SLOT_LV = 100;
+    /// <summary>
+    /// 插槽和角色的最大等级差
+    /// </summary>
+    public const int MAX_LV_GAP_BETWEEN_SLOT_AND_ROLE = 4;
+    /// <summary>
+    /// 角色装备插槽等级大于角色等级-5的最小数量
+    /// </summary>
+    public const int MIN_NUM_OF_ROLE_LV_BIGGER_SLOT_LV_THAN_NEGATIVE_5 = 4;
+    public const int TEST_DITAMIN = 999999;
```
[the remark this answers is not in the document]
Comment (lmangit) [reply Assets/Src/Module/MainPlayerRole/RoleLevelModule.cs]: 已经删了的

File: Assets/Src/Module/MainPlayerRole/view/ComRoleInfoLogic.cs [hunk 4]
```diff
@@ -28,33 +32,127 @@ protected override void OnAdd()
...
+
+    private void UpdateAttr()
+    {
+        Bian.EntityProfile profile = DataManager.MainPlayer.RoleData.Profile;
+        SetAttr("hp", profile.HpCurrent.ToString(), profile.HpLimit.ToString());
+        SetAttr("hpRecovery", profile.HpRecovery.ToString());
+        SetAttr("atk", profile.Att.ToString());
+        SetAttr("atkSpeed", profile.AttSpeed.ToString("f2") + "%");
+        SetAttr("def", profile.Def.ToString());
+        SetAttr("critRate", ((float)profile.CritRate / 10).ToString("f2") + "%");
```
[the remark this answers is not in the document]
Comment (lmangit) [reply Assets/Src/Module/MainPlayerRole/view/ComRoleInfoLogic.cs:103]: 这个要我去问下策划，我看白鹭那边代码是这样的

File: Assets/Src/Module/MainPlayerRole/view/ComSkillInfoLogic.cs (hunk 1 above)
Comment (lmangit) [reply Assets/Src/Module/MainPlayerRole/view/ComSkillInfoLogic.cs:83]: 好

File: Assets/Src/Module/MainPlayerRole/view/ComSlotInfoLogic.cs (hunk 2 above)
Comment (lmangit) [reply Assets/Src/Module/MainPlayerRole/view/ComSlotInfoLogic.cs:62]: ok

**Review by qqoo6789 — APPROVED** [review]

**Pull request merged**